.githooks/hooks/pre-commit

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+LC_ALL=C
+
+local_branch="$(git rev-parse --abbrev-ref HEAD)"
+
+valid_branch_regex="^(feat|spike|fix|test|refactor|renovate)\/CIV-[0-9]+|renovate\/"
+
+message="The branch names in this project must adhere to this contract: $valid_branch_regex. You should rename your branch to a valid name and try again. Eg:feat/CIV-1234, fix/CIV-1234-something, renovate/something"
```

Comment:
I don't think people will necessarily understand the regex you put here.
Make it more explicit and mention the allowed prefixes one by one.